src/main/java/org/ajoberstar/gradle/git/publish/tasks/GitPublishCommit.java

```diff
@@ -51,8 +59,15 @@ public void commit() {
     if (git.status().isClean()) {
       setDidWork(false);
     } else {
-      git.commit(op -> op.setMessage(getMessage().get()));
+      git.commit(this::commit);
```

Comment:
The `this::commit` (due to the name overlap between the public and private `commit` methods makes this a little hard to read. I'd probably just inline the logic:

```java
git.commit(op -> {
  // ...
});
```

Reply:
I prefer this version, because you end up with three levels of indentation without it. I've inlined it nevertheless.